lib/spack/spack/test/util/executable.py

```diff
@@ -36,21 +36,31 @@ def test_read_unicode(tmpdir, working_env):
         filter_shebangs_in_directory('.', [script_name])
 
         # read the unicode back in and see whether things work
-        script = ex.Executable('./%s' % script_name)
+        if sys.platform == 'win32':
+            script = ex.Executable('%s %s' % (sys.executable, script_name))
+        else:
+            script = ex.Executable('./%s' % script_name)
+
         assert u'\xc3' == script(output=str).strip()
 
 
 def test_which_relative_path_with_slash(tmpdir, working_env):
-    tmpdir.ensure('exe')
-    path = str(tmpdir.join('exe'))
+    tmpdir.ensure("exe")
+    path = str(tmpdir.join("exe"))
+
     os.environ['PATH'] = ''
 
     with tmpdir.as_cwd():
-        no_exe = ex.which('./exe')
+        no_exe = ex.which('.{0}exe'.format(os.path.sep))
         assert no_exe is None
-
-        fs.set_executable(path)
-        exe = ex.which('./exe')
+        if sys.platform == "win32":
+            # For Windows, need to create files with .exe after any assert is none tests
+            tmpdir.ensure("exe.exe")
```

Comment:
Can you do this path creation at the beginning of the test? Like instead of having:

```
def test_which_relative_path_with_slash(tmpdir, working_env):
    tmpdir.ensure("exe")
```

You do

```
def test_which_relative_path_with_slash(tmpdir, working_env):
    if is_windows:
        tmpdir.ensure(exe.exe)
    else:
        tmpdir.ensure(exe)
```

I think in general that consolidating the setup will make the tests more readable, where that is possible.